app/controllers/status_messages_controller.rb

```diff
@@ -62,7 +62,9 @@ def create
         @status_message.poll.poll_answers.build(:answer => poll_answer)
       end
     end
-
+    if params[:event_name].present?
+      @status_message.build_event(:name => params[:event_name], :location => params[:event_location], :date => params[:event_date])
```

Comment:
Use Ruby 1.9 style hash syntax. (`name: params[:event_name]`), location: params[:event_location...`). Maybe consider passing this as nested parameters, so you can just `.build_event(event_params)` with
```ruby
def event_params
  params.require(:event).permit(:name, :location, :date)
end
```